core/txpool/txpool.go

```diff
@@ -706,17 +731,34 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (replaced bool, err e
 			overflowedTxMeter.Mark(1)
 			return false, ErrTxPoolOverflow
 		}
+
+		// If the new transaction is a future transaction it should never churn pending transactions
+		if pool.isFuture(from, tx) {
+			for _, dropTx := range drop {
+				dropSender, _ := types.Sender(pool.signer, dropTx)
+				if list := pool.pending[dropSender]; list != nil && list.Overlaps(dropTx) {
+					// Add all transactions back to the priced queue
+					for _, d := range drop {
+						pool.priced.urgent.Push(d)
```

Comment:
Not sure this is correct. It is usually done like this 
```
	heap.Push(&l.urgent, tx)
```
I'm not sure what happens when you `Push` directly like that